network/src/lib.rs

```diff
@@ -75,8 +83,34 @@ where
 		header: &B::Header,
 		mut data: &[u8],
 	) -> Result<Validation, Box<dyn std::error::Error + Send>> {
-		// If no data is provided the announce is valid.
+		let runtime_api = self.polkadot_client.runtime_api();
+		let polkadot_info = self.polkadot_client.info();
+
 		if data.is_empty() {
+			// Check if block is one higher than best
```

Comment:
You need a justification if this is a new best block.

Reply:
like this? https://github.com/paritytech/cumulus/pull/105/commits/2f9131c563d74a36b7e34f989f334e0eac8288d1